### ISSUE TITLE
Revisit distributed search tree benchmark

### ISSUE THREAD

**Issue by dalg24** [arborx/ArborX#57]
Catching exceptions and still returning error code zero
```
13/14 Testing: ArborX_DistributedTree_Example
13/14 Test: ArborX_DistributedTree_Example
Command: "/opt/openmpi/bin/mpiexec" "-n" "8" "--allow-run-as-root" "./ArborX_DistributedTree.exe" "--benchmark_color=no"
Directory: /var/jenkins/workspace/ArborX-experimental_PR-55/build/examples/distributed_tree_driver
"ArborX_DistributedTree_Example" start time: May 10 02:57 UTC
Output:
----------------------------------------------------------
Kokkos::OpenMP::initialize WARNING: OMP_PROC_BIND environment variable not set
  In general, for best performance with OpenMP 4.0 or better set OMP_PROC_BIND=spread and OMP_PLACES=threads
  For best performance with OpenMP 3.1 set OMP_PROC_BIND=true
  For unit testing set OMP_PROC_BIND=false
Kokkos::OpenMP::initialize WARNING: OMP_PROC_BIND environment variable not set
  In general, for best performance with OpenMP 4.0 or better set OMP_PROC_BIND=spread and OMP_PLACES=threads
  For best performance with OpenMP 3.1 set OMP_PROC_BIND=true
  For unit testing set OMP_PROC_BIND=false
Kokkos::OpenMP::initialize WARNING: OMP_PROC_BIND environment variable not set
  In general, for best performance with OpenMP 4.0 or better set OMP_PROC_BIND=spread and OMP_PLACES=threads
  For best performance with OpenMP 3.1 set OMP_PROC_BIND=true
  For unit testing set OMP_PROC_BIND=false
Kokkos::OpenMP::initialize WARNING: OMP_PROC_BIND environment variable not set
  In general, for best performance with OpenMP 4.0 or better set OMP_PROC_BIND=spread and OMP_PLACES=threads
  For best performance with OpenMP 3.1 set OMP_PROC_BIND=true
  For unit testing set OMP_PROC_BIND=false
Kokkos::OpenMP::initialize WARNING: OMP_PROC_BIND environment variable not set
  In general, for best performance with OpenMP 4.0 or better set OMP_PROC_BIND=spread and OMP_PLACES=threads
  For best performance with OpenMP 3.1 set OMP_PROC_BIND=true
  For unit testing set OMP_PROC_BIND=false
Kokkos::OpenMP::initialize WARNING: OMP_PROC_BIND environment variable not set
  In general, for best performance with OpenMP 4.0 or better set OMP_PROC_BIND=spread and OMP_PLACES=threads
  For best performance with OpenMP 3.1 set OMP_PROC_BIND=true
  For unit testing set OMP_PROC_BIND=false
Kokkos::OpenMP::initialize WARNING: OMP_PROC_BIND environment variable not set
  In general, for best performance with OpenMP 4.0 or better set OMP_PROC_BIND=spread and OMP_PLACES=threads
  For best performance with OpenMP 3.1 set OMP_PROC_BIND=true
  For unit testing set OMP_PROC_BIND=false
Kokkos::OpenMP::initialize WARNING: OMP_PROC_BIND environment variable not set
  In general, for best performance with OpenMP 4.0 or better set OMP_PROC_BIND=spread and OMP_PLACES=threads
  For best performance with OpenMP 3.1 set OMP_PROC_BIND=true
  For unit testing set OMP_PROC_BIND=false
processor 1 caught a std::exception: unrecognised option '--benchmark_color=no'
processor 7 caught a std::exception: unrecognised option '--benchmark_color=no'
processor 0 caught a std::exception: unrecognised option '--benchmark_color=no'
processor 4 caught a std::exception: unrecognised option '--benchmark_color=no'
processor 5 caught a std::exception: unrecognised option '--benchmark_color=no'
processor 3 caught a std::exception: unrecognised option '--benchmark_color=no'
processor 2 caught a std::exception: unrecognised option '--benchmark_color=no'
processor 6 caught a std::exception: unrecognised option '--benchmark_color=no'
<end of output>
Test time =   2.99 sec
----------------------------------------------------------
Test Passed.
"ArborX_DistributedTree_Example" end time: May 10 02:57 UTC
"ArborX_DistributedTree_Example" time elapsed: 00:00:02
----------------------------------------------------------
```

**Comment by aprokop** [comment]
Will not fix, works as intended 😄

**Comment by masterleinad** [comment]
I don't see this anymore passing
```
--------------------------------------------------------------------------
mpiexec detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[34894,1],3]
  Exit code:    1
--------------------------------------------------------------------------
```

**Comment by aprokop** [comment]
I believe this was fixed in ff24c628b75613702b3fa1ed6778f36737687381.
